main.go

```diff
@@ -113,13 +116,23 @@ func makeAndStartNode(ds ds.Batching, addr string, relay bool, bucketSize int) (
 	}
 
 	go func() {
+		if limiter != nil {
+			limiter <- struct{}{}
+		}
 		err = h.Connect(context.Background(), boostrapper())
 		if err != nil {
 			panic(err)
 		}
 
 		time.Sleep(time.Second)
+
+		d.BootstrapOnce(context.Background(), dht.BootstrapConfig{Queries: 4})
```

Comment:
Can we move all the `context.Background()`s out and have a single root?

Reply:
Can we do it as separate PR? I want to get this merged so we can figure out how many nodes we can run on single box.

Reply:
Sure